facilitator/src/kubernetes.rs

```diff
@@ -0,0 +1,138 @@
+use anyhow::{anyhow, Result};
+
+use kube::{
+    api::{Api, ListParams},
+    Client,
+};
+
+use k8s_openapi::{api::core::v1::Secret, Metadata};
+
+use tokio::runtime::Runtime;
+
+/// Definition of a namespaced Kubernetes API implementation
+#[derive(Debug)]
+pub struct Kubernetes {
```

Comment:
This should be called something like `KubernetesClient` to make it clear we're not implementing a Kubernetes API server here.